hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+
+		// Creating Start JointTrajectoryPt from start joint pose
+		descartes_core::TrajectoryPtPtr start_joint_point = descartes_core::TrajectoryPtPtr(
+		new descartes_trajectory::JointTrajectoryPt(start_pose));
+
+		// Creating End JointTrajectoryPt from end joint pose
+		descartes_core::TrajectoryPtPtr end_joint_point = descartes_core::TrajectoryPtPtr(
+		new descartes_trajectory::JointTrajectoryPt(end_pose));
+
+		// Modifying start and end of the trajectory.
+		descartes_traj_input_[0] = start_joint_point;
+		descartes_traj_input_[descartes_traj_input_.size() - 1 ] = end_joint_point;
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to find closest joint pose to seed pose at the start or end of trajectory");
+		exit(-1);
+		}
+
+		//TOCHECK Should the planners be initialized/modified in here instead of init()
+		//as the RobotPtr might have changed in its internal parameters
+
+		if(config_.descartes_planner_type == "dense")
+		{
+			bool succeeded = dense_planner_.planPath(descartes_traj_input_);
+			if (succeeded)
+			{
+			ROS_INFO_STREAM("Valid path was found using the Descartes dense planner");
+			}
+			else
+			{
+			ROS_ERROR_STREAM("Could not solve for a valid path using the Descartes dense planner");
+			exit(-1);
+			}
+
+			succeeded = dense_planner_.getPath(descartes_traj_output_);
+
+			if(!succeeded || descartes_traj_output_.empty())
+			{
+			ROS_ERROR_STREAM("Failed to retrieve robot path");
+			}
+		}
+
+		if(config_.descartes_planner_type == "sparse")
+		{
+			bool succeeded = sparse_planner_.planPath(descartes_traj_input_);
+			if (succeeded)
+			{
+			ROS_INFO_STREAM("Valid path was found using the Descartes sparse planner");
+			}
+			else
+			{
+			ROS_ERROR_STREAM("Could not solve for a valid path using the Descartes sparse planner");
+			exit(-1);
+			}
+
+			succeeded = sparse_planner_.getPath(descartes_traj_output_);
+
+			if(!succeeded || descartes_traj_output_.empty())
+			{
+			ROS_ERROR_STREAM("Failed to retrieve robot path");
+			}
+		}
+		
+		ROS_INFO_STREAM("Task '"<<__FUNCTION__<<"' completed");
+	}
+
+	void HybridPlanner::runPath()
+	{
+		moveit_group_ptr_->setPlannerId(PLANNER_ID);//TOCHECK should be in init(). Also the config_ param or not doubt. Check header file. 
+		overall_robot_traj_.getRobotTrajectoryMsg(moveit_robot_traj_);
+
+		// creating goal joint pose to start of the path
+		// std::vector<double> seed_pose(robot_model_ptr_->getDOF());
+  		std::vector<double> start_pose;
+  		descartes_core::TrajectoryPtPtr first_point_ptr = path[0];
+		first_point_ptr->getNominalJointPose(config_.seed_pose,*descartes_robot_model_ptr_,start_pose);
+
+		// moving arm to joint goal
+		moveit_group_ptr_->setJointValueTarget(start_pose);
+		moveit_group_ptr_->setPlanningTime(config_.plan_time);
+		moveit_msgs::MoveItErrorCodes result = moveit_group_ptr_->move();
+		if(result.val != result.SUCCESS)
+		{
+		ROS_ERROR_STREAM("Move to start joint pose failed");
+		exit(-1);
+		}
+
```

Comment:
This function is incomplete. 
So by now, after calling `appendFreeSpaceSegment()` and `appendDescartesCartPointSegment()` multiple times, followed by a `planProcessPath()`, we have the freeSpaceTrajectories in `robot_trajectory::RobotTrajectory overall_robot_traj_` and the processTrajectories in `descartes_traj& descartes_traj_output_`
Note : 
`typedef moveit_msgs::RobotTrajectory moveit_robot_traj;`
`typedef std::vector<descartes_core::TrajectoryPtPtr> descartes_traj;` (#L41 in header file)

Then I do a `overall_robot_traj_.getRobotTrajectoryMsg(moveit_robot_traj_);` in the `runPlan()` method above.